src/executor/mod.rs

```diff
@@ -1,5 +1,7 @@
 //! Types which can take a boxed closure and execute it, preferably in the background.
 
+#![cfg_attr(feature="clippy", allow(boxed_local))]
```

Comment:
Yeah, this is the exact case: https://github.com/Manishearth/rust-clippy/issues/1123

But it is kind of a duplicate of #916.